src/Types/BigIntType.php

```diff
@@ -54,7 +54,7 @@ public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int
 
         if (
             ($value > PHP_INT_MIN && $value < PHP_INT_MAX)
-            || $value === (string) (int) $value
+            || $value === (string) @(int) $value
```

Comment:
What if the whole condition was replaced with `$value >= PHP_INT_MIN && $value <= PHP_INT_MAX`? Are there other values that would still only be covered by the second part of the condition?

Reply:
We have to be very careful about this because a large numeric string cast to int will be converted to `PHP_INT_MAX`. But yeah, I think you're on the right track here.